docs/docs/guides/web3_plugin_guide/index.md

```diff
@@ -1,50 +1,334 @@
-- [Plugin List](https://web3js.org/plugins)
 
+- You can find all the web3 plugins🧩 [here](https://web3js.org/plugins) 
+
+- To list your web3 plugin🧩 into the web3js.org/plugins page, you can submit a PR [here](https://github.com/web3/web3js-landing/blob/main/src/pluginList.ts)
+
+## Create a plugin
+
+```javascript
+//1. import the `Web3PluginBase` module
+import { Web3PluginBase } from "web3";
+
+//2. Create a Class extending the `Web3Pluginbase`
+class MyPlugin extends Web3PluginBase {
+    
+    //3. Add a name to the plugin
+    pluginNamespace = "pluginExample"; 
+
+    //4. Create any methods with your desired functionality
+    async doSomething(){
+        console.log("Hello web3!");
+        //send transactions
+        //initialize contracts
+        //deploy or interact with contracts
+        //add your own library, logic or functionality
+        //much more...
+    }
+}
+
+module.exports = MyPlugin;
+```
+
+## Use a plugin
+
+```javascript
+//1. import the plugin and web3 module
+import { Web3 } from "web3";
+import MyPlugin from "./plugin";
+
+//2. Initialize the web3 instance
+const web3 = new Web3("https://eth.llamarpc.com");
+
+//3. Register plugin to add the current Web3Context
+web3.registerPlugin(new MyPlugin());
+
+//4. Use the plugin methods
+web3.pluginExample.doSomething();
+//--> Hello web3!
+```
+
+## Initialize a provider for the plugin
+
+There are 2 ways to initialize a provider when you are using a plugin, the [first one](#set-a-provider-within-the-plugin) is within the plugin (internally using the `this` object and `setProvider()` method). And the [second one](#register-plugin), is done by the final user by using the `registerPlugin()` method.
+
+### Register plugin (External - Final user)
+
+When the final user (developer) is using the plugin, they must write the following:
+
+```js
+//Step #1: Import the plugin and web3 module
+import { Web3 } from "web3";
+import  MyPlugin from "./plugin";
+
+//Step #2: Initialize the web3 instance
+const web3 = new Web3("https://eth.llamarpc.com");
+
+//Step #3: Register plugin to add the current Web3Context
+web3.registerPlugin(new MyPlugin());
+```
+
+And once the plugin is registered in step #3 by using `registerPlugin(new Plugin)`, the provider in the plugin is automatically set to the provider initialized in `Step #2` (in this case: `https://eth.llamarpc.com`).
+
+### Set a provider within the plugin (web3Context)
+
+If you want to initialize the provider inside the plugin you can use the `this` object, but anyway the user must initialize a provider when [registering the plugin](#register-plugin-external---final-user):
+
+```js
+//Step #1: Import modules
+import { Web3PluginBase, eth } from "web3";
+
+//Step #2: Create class
+class MyPlugin extends Web3PluginBase {
+  pluginNamespace = "myPlugin";
+
+//Step #3: Set providers
+  async getChainIdMainnet() {
+    this.setProvider("https://eth.llamarpc.com") //or any other RPC
+    return await eth.getChainId(this)
+  }
```

Comment:
these setprovider will only work for rest of lib if its context is applied via link override like: 
https://github.com/ChainSafe/web3-plugin-chainstack/blob/fbe7b2376ca8623db3801b41f96e4f0fb3b8ef2d/src/chainstack-plugin.ts#L43

and provider is set for parent context like:
https://github.com/ChainSafe/web3-plugin-chainstack/blob/fbe7b2376ca8623db3801b41f96e4f0fb3b8ef2d/src/chainstack-plugin.ts#L87